sos/plugins/__init__.py

```diff
@@ -1256,27 +1256,109 @@ def collect_cmd_output(self, cmd, suggest_filename=None,
         )
 
     def exec_cmd(self, cmd, timeout=300, stderr=True, chroot=True, runat=None,
-                 env=None, binary=False, pred=None):
+                 env=None, binary=False, pred=None, container=False):
         """Execute a command right now and return the output and status, but
         do not save the output within the archive.
 
         Use this method in a plugin's setup() if command output is needed to
         build subsequent commands added to a report via add_cmd_output().
         """
+        _default = {'status': None, 'output': ''}
         if not self.test_predicate(cmd=True, pred=pred):
-            return {
-                'status': None,
-                'output': ''
-            }
+            return _default
 
         if chroot or self.commons['cmdlineopts'].chroot == 'always':
             root = self.sysroot
         else:
             root = None
 
+        if container:
+            if self._get_container_runtime() is None:
+                self._log_info("Cannot run cmd '%s' in container %s: no "
+                               "runtime detected on host." % (cmd, container))
+                return _default
+            if self.container_exists(container):
+                cmd = self.fmt_container_cmd(container, cmd)
+            else:
+                self._log_info("Cannot run cmd '%s' in container %s: no such "
+                               "container is running." % (cmd, container))
+
```

Comment:
Shall not we call `return _default` here? (well, current behaviour will call docker/podman cmd on non-existing container that works "well" also).

Reply:
Hmm.. it might be valuable to let execute that command and get something like:

    Error: error getting image "3848ec06ffb6": unable to find a name and tag match for 3848ec06ffb6 in repotags: no such image

No preferences from my side here.

Reply:
Hmm, fair point on executing it anyway and collecting the failure. I'll think this over a bit more.